includes/class-wc-stripe-customer.php

```diff
@@ -334,6 +334,38 @@ public function set_default_source( $source_id ) {
 		return false;
 	}
 
+	/**
+	 * Retrieves the default payment method either from the WC or Stripe customer.
+	 *
+	 * @since 4.3.1
+	 * @return object|null Either a basic source object, or null.
+	 */
+	public function get_default_source() {
+		if ( ! $this->get_id() ) {
+			return null;
+		}
+
+		// First, try to load the default source from the WooCommerce customer object.
+		if ( ! empty( $this->user_id ) ) {
+			$default_token = WC_Payment_Tokens::get_customer_default_token( $this->user_id );
+
+			if ( ! empty( $default_token ) && 'stripe' === $default_token->get_gateway_id() ) {
+				$source = WC_Stripe_API::request( array(), 'sources/' . $default_token->get_token(), 'GET' );
+				if ( empty( $source->error ) && $this->id === $source->customer ) {
+					return $source;
+				}
+			}
+		}
+
+		// Second, try to load the default source based on the user.
```

Comment:
Why is it better to try to get the customer's source from a previous order vs just getting the default source right away?

Reply:
My initial thinking was that it's possible not to have a WooCommerce user, associated with the Stripe customer. That way thinking was caused by the fact that merchants can manually enter a customer ID when creating a subscription.

So the way this works within `get_default_source`:

1. Check for `! empty( $this->user_id )`. If the closure is entered, this means that the merchant choose an existing WooCommerce user.
2. If the merchant entered the customer ID manually, then the customer ID might not be associated with the WooCommerce user.